instrumentation-genai/opentelemetry-instrumentation-google-genai/src/opentelemetry/instrumentation/google_genai/generate_content.py

```diff
@@ -129,21 +131,60 @@ def _determine_genai_system(models_object: Union[Models, AsyncModels]):
     return _get_gemini_system_name()
 
 
-def _get_config_property(
-    config: Optional[GenerateContentConfigOrDict], path: str
-) -> Any:
+def _to_dict(value: object):
+    if isinstance(value, dict):
+        return value
+    if hasattr(value, "model_dump"):
+        return value.model_dump()
+    return json.loads(json.dumps(value))
+
+
+def _add_request_options_to_span(
+    span, config: Optional[GenerateContentConfigOrDict]
+):
     if config is None:
-        return None
-    path_segments = path.split(".")
-    current_context: Any = config
-    for path_segment in path_segments:
-        if current_context is None:
-            return None
-        if isinstance(current_context, dict):
-            current_context = current_context.get(path_segment)
-        else:
-            current_context = getattr(current_context, path_segment)
-    return current_context
+        return
+    span_context = span.get_span_context()
+    if not span_context.trace_flags.sampled:
+        # Avoid potentially costly traversal of config
+        # options if the span will be dropped, anyway.
+        return
+    # Automatically derive attributes from the contents of the
+    # config object. This ensures that all relevant parameters
```

Comment:
Is it possible that this blows up the attributes size or accidentally includes sensitive data with the dynamic approach? For reference, `http.request.header.<key>` is opt in for this kind of thing https://opentelemetry.io/docs/specs/semconv/http/http-spans/#http-client

Reply:
The safest approach is to opt-in every field because the config could include sensitive information in newly added fields  in the future and even updating the filter list will still result in previous versions of this opentelemetry integration referencing those fields.

Some headers may be useful for tracking. For example, provisioned throughput: https://cloud.google.com/vertex-ai/generative-ai/docs/use-provisioned-throughput

Reply:
OK. I have updated this to use an allow list for determining which dynamic keys to include.